### PR TITLE
Feature/translated consent docs

### PR DESCRIPTION
**Pull request by EdoStorm96**
Adresses issue 401. Ask whether the consent forms will need to be translated eventually. Added new attributes to the proposal model and added a new page in the process, before uploading the consent forms. The implementation has been discussed with Desiree.

### CODE REVIEW

**Review by miggol — CHANGES_REQUESTED** [review]
Good job thinking of the pdf and diff pages as well when adding the attributes.

File: proposals/utils/proposal_utils.py
Comment (EdoStorm96) [reply proposals/utils/proposal_utils.py]: I did leave line 107 be, because the consent_docs_url is a child of the consent docs_url, so these lead to purposely different urls. I did implement all you other suggestions though.

File: proposals/utils/proposal_utils.py
Comment (miggol) [inline proposals/utils/proposal_utils.py:101]: Some of these lines are very long. Wrapping makes them hard to read and they hide the structure of the function call. Compare:

    translated_docs_url = AvailableURL(title=_('Vertaling'), url=reverse('proposals:translated', args=(proposal.pk, )))
    consent_url = AvailableURL(title=_('Formulieren'), children=[translated_docs_url, consent_docs_url])

With the following:

    translated_docs_url = AvailableURL(
        title=_('Vertaling'),
        url=reverse(
            'proposals:translated',
            args=(proposal.pk,),
        )
    )
    consent_url = AvailableURL(
        title=_('Formulieren'),
        children=[
            translated_docs_url,
            consent_docs_url
        ]
    )

I much prefer it when similar code has a similar "shape"!

I know much of the Ethics doesn't follow these guidelines very well, but we try to write all new code in line with PEP 8. If you have a small font size or wide monitor, consider turning on an IDE warning when lines get too long.

File: proposals/forms.py
Comment (miggol) [inline proposals/forms.py:655]: These lines are also too long, and not my preference in indentation. You're jumping from an indent of 12 to 48 lines! I would much prefer

    self.add_error(
        'translated_forms_languages',
        _('Vul in in welke talen de formulieren worden vertaald.'),
    )

This also saves you from breaking up the string.

File: proposals/forms.py
Comment (miggol) [inline proposals/forms.py:635]: So there's a bit of a weird interaction/bug with Django and how we build this multiple page form. If you're on a page like TranslatedConsentForms and there's an error in your input, you actually can't navigate away from it with the forward or next buttons. That's because those buttons try to save the input along the way, but that causes a Form error and Django says no. 

For this reason, SoftValidationMixin exists. We use it on proposal forms to solve this issue. It allows validation to continue and output its errors, but doesn't hard-prevent saving or form submission.

TranslatedConsentForms should use sfot validation as well, and it should mention both of its fields in `TranslatedConsentForms._soft_validation_fields`. But try to recreate the bug first so you get what I'm talking about

**Review by miggol — CHANGES_REQUESTED** [review]
It's become time to introduce you to `/proposals/utils/validate_proposal.py`.

When you're on the final page of the proposal form (ProposalSubmit), the page checks if there are any errors on the previous pages and then mentions them to you. Depending on those errors it allows you to submit or not.

Because TranslatedConsentForms isn't yet included in `validate_proposal.py`, errors from it don't show up or prevent submission.

Take a look and see if you can add TranslatedConsentForms to the validation yourself, but please ask for help at any point because proposal validation is a bit weird.

File: proposals/utils/validate_proposal.py
Comment (miggol) [inline proposals/utils/validate_proposal.py:223]: I think you forgot to `makemessages` after adding this string because it isn't in `django.po`

**Review by miggol — APPROVED** [review]
Great! Assuming the single missed string will be fixed soon enough I'm approving this PR.

Please do not merge into develop until we've merged migrations together.

**Comment by miggol** [comment]
Migrations seem to be fine, so this can be merged! (once translations fixed)

**Pull request merged**
